modules/020-deckhouse/hooks/update_deckhouse_image.go

```diff
@@ -301,10 +313,20 @@ func releaseChannelUpdate(input *go_hook.HookInput, releases []deckhouseReleaseU
 		// "Deployed" shows only Actual (current) release. All previous releases are marked as Outdated
 		// It's much more comfortable to observe DeckhouseReleases like this because by default they are sorted by Name
 		// and sometimes it's a bit weird for semver names. This statuses shows you the real view of releases
-		case v1alpha1.PhaseOutdated:
+		case v1alpha1.PhaseOutdated, v1alpha1.PhaseSuspended:
 			// pass
 
 		case v1alpha1.PhasePending:
+			// skip suspended releases
+			if rl.HasSuspendAnnotation {
+				sp := statusPatch{
+					Phase:          v1alpha1.PhaseSuspended,
+					TransitionTime: now,
+				}
+				input.PatchCollector.MergePatch(sp, "deckhouse.io/v1alpha1", "DeckhouseRelease", "", rl.Name, object_patch.WithSubresource("/status"))
```

Comment:
Looked through briefly and I think I missed something. Could you elaborate more on why we need both the annotation and the status field?

Reply:
Annotation is placed by a hook which checks the release image. It marks a release as "should be suspended".
And this hook takes marked release and move it to a suspended state to make it more obvious for user experience. Also it makes sense for a release loop